tests/regression/runner.py

```diff
@@ -303,14 +305,87 @@ def test_send_transfer(self):
         rand_addr = gen_rand_trytes(81)
         for i in range(TIMES_TOTAL):
             start_time = time.time()
-            post_data = "{\"value\":0,\"message\":\"" + str(
-                rand_msg) + "\",\"tag\":\"" + str(
-                    rand_tag) + "\",\"address\":\"" + str(rand_addr) + "\"}"
-            API("/transaction/", post_data=post_data, delay=False)
+            post_data = {
+                "value": 0,
+                "message": rand_msg,
+                "tag": rand_tag,
+                "address": rand_addr
+            }
+            post_data_json = json.dumps(post_data)
+            response.append(API("/transaction/", post_data=post_data_json))
             time_cost.append(time.time() - start_time)
 
         eval_stat(time_cost, "send transfer")
 
+    def test_find_transactions_by_tag(self):
+        logging.debug(
+            "\n================================find transactions by tag================================"
+        )
+        # cmd
+        #    0. 27 trytes tag
+        #    1. 20 trytes tag
+        #    2. 30 trytes tag
+        #    3. unicode trytes tag
+        #    4. Null trytes tag
+        rand_tag_27 = gen_rand_trytes(27)
+        rand_tag_20 = gen_rand_trytes(20)
+        rand_tag_30 = gen_rand_trytes(30)
+        test_cases = [rand_tag_27, rand_tag_20, rand_tag_30, "半導體絆倒你", None]
+
+        rand_msg = gen_rand_trytes(30)
+        rand_addr = gen_rand_trytes(81)
+        transaction_response = []
+        for i in range(3):
+            post_data = {
+                "value": 0,
+                "message": rand_msg,
+                "tag": test_cases[i],
+                "address": rand_addr
+            }
+            post_data_json = json.dumps(post_data)
+            transaction_response.append(
+                API("/transaction/", post_data=post_data_json))
+        if DEBUG_FLAG == True:
+            for i in range(len(transaction_response)):
+                logging.debug("find transactions by tag i = " + str(i) +
+                              ", tx_res = " + transaction_response[i][0] +
+                              ", status code = " + transaction_response[i][1])
+
+        response = []
+        for t_case in test_cases:
+            logging.debug("testing case = " + repr(t_case))
+            if t_case != None:
+                response.append(API("/tag/", get_data=(t_case + "/hashes")))
+            else:
+                response.append(API("/tag/", get_data="/hashes"))
+
+        if DEBUG_FLAG == True:
```

Comment:
I think the design of `DEBUG_FLAG` is really redundant. Perhaps it can just integrate in below?

Reply:
I will just remove the ` if DEBUG_FLAG == True:`